docs/concepts.rst

```diff
@@ -842,61 +842,135 @@ values by "case" instead of by entity.
-is usually a sensible number, but it can also be set directly:
+[#workers]_, each of which can work on one value at a time. By default, Bionic
+will create one worker process for each CPU on your machine. This is usually a
+sensible number, but it can also be set directly:
 
 .. code-block:: python
 
     builder.set("core__parallel_execution__worker_count", 8)
 
+.. [#workers] The pool of workers is managed by
+  `Loky <https://loky.readthedocs.io/en/stable/>`_,
+  which is built on Python's
+  `multiprocessing <https://docs.python.org/3.8/library/multiprocessing.html>`_ module.
+  The pool is global and reusable, so it should only need to be initialized once in
+  the lifetime of the main process.
+
+Google AI Platform Execution
+............................
+
+.. versionadded:: 0.9.0
+
+Bionic can make use of compute resources in the cloud by sending entities to
+`Google AI Platform (AIP) <https://cloud.google.com/ai-platform>`_ for
+computation.
+
+Using this feature requires specifying the GCP project as well as a docker image
+in Google Container Registry containing the same Python environment (runtime and
+libraries) as the one used to invoke Bionic.
+
+.. code-block:: python
+
+    builder.set('core__aip_execution__enabled', True)
+    builder.set('core__aip_execution__gcp_project_name', 'my-gcp-project')
+    builder.set('core__aip_execution__docker_image_name', 'bionic:latest')
+
+Additionally, :ref:`Google Cloud Storage <google_cloud_storage_anchor>` must be
+enabled.
+
+Entities marked with :func:`@aip_task_config <bionic.aip_task_config>` decorator
+will use AIP for computation. It takes in a `Compute Engine machine type identifier <https://cloud.google.com/ai-platform/training/docs/machine-types#compute-engine-machine-types>`_.
+Such entities must be serializable and cannot be marked with :func:`@persist(False) <bionic.persist>`.
+
+.. code-block:: python
+
+    @builder
+    @bionic.aip_task_config('n1-standard-4')
+    def x():
+      return 1
+
+Please refer to Docker documentation on how to create a docker image and push
+it to Google Container Registry. Here is an example script to build and push a
+docker image mirroring your local Python environment, assuming that all the
+modules are installed through ``pip``.
+
+.. code-block:: bash
+
+    pip freeze > requirements.txt
+
+    cat > Dockerfile <<EOF
+    FROM python:3.8
+    COPY requirements.txt requirements.txt
+    RUN pip install -r requirements.txt
+    EOF
+
+    docker build -t gcr.io/my-gcp-project/bionic:latest .
+    docker push gcr.io/my-gcp-project/bionic:latest
+
+.. _parallel-caveats:
+
+Caveats
+.......
+
+There is overhead for doing computation in external processes. The overhead is
+significant for Google AI Platform because it may take a few minutes to spin up
+a new job instance. Parallelism is beneficial if the computation time far
+exceeds the overhead.
+
+Also, Bionic can only start computing a value once all its dependencies are
```

Comment:
I would start this paragraph with the most important point, which is that the total speedup depends on the structure of your graph.